### ISSUE TITLE
More flexible type specification

### ISSUE THREAD

**Issue by aforren1** [aforren1/toon#68]
```python
import numpy as np

x = np.array([3,2,1])
# get ctypes version of dtype
np.ctypeslib._typecodes[x.__array_interface__['typestr']]
```

**Comment by aforren1** [comment]
Map from `np.ctypeslib._typecodes` (with c_char added):
```python
ctypes_map = np.ctypeslib._typecodes.copy()
ctypes_map['|S1'] = ctypes.c_char
```

**Comment by aforren1** [comment]
So it goes any (python type, dtype, ctype) -> dtype -> ctype

**Comment by aforren1** [comment]
Closed by 8c8c813b50ac0311f9f0b48d7cd8bd3f5e934065
